### PR TITLE
Fix: Add `RequiredSignatures` to the `BranchProtectionSettingsUpdate` Type

### LINKED ISSUE #2855: [BUG]: BranchProtectionSettingsUpdate type is missing a bool property for RequiredSignatures.
### What happened?

There is no options to set `required_signatures` to true or false with the BranchProtectionSettingsUpdate.

```json
{
  "url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection",
  "required_status_checks": {
    "url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/required_status_checks",
    "strict": true,
    "contexts": [
      "continuous-integration/travis-ci"
    ],
    "contexts_url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/required_status_checks/contexts",
    "checks": [
      {
        "context": "continuous-integration/travis-ci",
        "app_id": null
      }
    ]
  },
  "restrictions": {
    "url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/restrictions",
    "users_url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/restrictions/users",
    "teams_url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/restrictions/teams",
    "apps_url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/restrictions/apps",
    "users": [],
    "teams": [],
    "apps": []
  },
  "required_pull_request_reviews": {
    "url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/required_pull_request_reviews",
    "dismiss_stale_reviews": true,
    "require_code_owner_reviews": true,
    "required_approving_review_count": 2,
    "require_last_push_approval": true,
    "dismissal_restrictions": {
      "url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/dismissal_restrictions",
      "users_url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/dismissal_restrictions/users",
      "teams_url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/dismissal_restrictions/teams",
      "users": [],
      "teams": [],
      "apps": []
    }
  },
  "required_signatures": {
    "url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/required_signatures",
    "enabled": false
  },
  "enforce_admins": {
    "url": "https://api.github.com/repos/octocat/hello-world/branches/main/protection/enforce_admins",
    "enabled": true
  },
  "required_linear_history": {
    "enabled": true
  },
  "allow_force_pushes": {
    "enabled": true
  },
  "allow_deletions": {
    "enabled": true
  },
  "block_creations": {
    "enabled": true
  },
  "required_conversation_resolution": {
    "enabled": true
  },
  "lock_branch": {
    "enabled": true
  },
  "allow_fork_syncing": {
    "enabled": true
  }
}
```
[(the above example is located here)](https://docs.github.com/en/rest/branches/branch-protection?apiVersion=2022-11-28#update-branch-protection)

### Versions

Octokit.net 9.1.0

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wesdevpro**
Resolves  #2855

----

### Before the change?

* No `RequiredSignatures` property present for `BranchProtectionSettingsUpdate` type which is supported by GitHub's rest APIs.

### After the change?

* Added `RequiredSignatures` property for the `BranchProtectionSettingsUpdate` type
* Updated all xUnnit tests for the `BranchProtectionSettingsUpdate` type to account for the new property (i.e `RequiredSignatures`)

### Pull request checklist
- [X] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [x] No

----

### CODE REVIEW

File: Octokit/Models/Request/BranchProtectionUpdate.cs [hunk 1]
```diff
@@ -122,6 +122,7 @@ public BranchProtectionSettingsUpdate(BranchProtectionRequiredStatusChecksUpdate
         public BranchProtectionSettingsUpdate(BranchProtectionRequiredStatusChecksUpdate requiredStatusChecks,
                                               BranchProtectionRequiredReviewsUpdate requiredPullRequestReviews,
                                               BranchProtectionPushRestrictionsUpdate restrictions,
+                                              bool requiredSignatures,
```
Comment (kfcampbell) [inline Octokit/Models/Request/BranchProtectionUpdate.cs:125]: Would you mind fixing these warnings by adding appropriate an appropriate docs comment?

Comment (wesdevpro) [reply Octokit/Models/Request/BranchProtectionUpdate.cs:125]: Warnings?

Comment (wesdevpro) [reply Octokit/Models/Request/BranchProtectionUpdate.cs:125]: > Would you mind fixing these warnings by adding appropriate an appropriate docs comment?

I will insert a new line with a comment on ln. 115

Comment (kfcampbell) [reply Octokit/Models/Request/BranchProtectionUpdate.cs:126]: Good, thanks!

**Review by kfcampbell — APPROVED** [review]

**Pull request merged**